lib/pbench/test/unit/server/test_schema.py

```diff
@@ -0,0 +1,151 @@
+import dateutil
+import pytest
+from typing import Callable
+
+from pbench.server.api.auth import Auth, UnknownUser
+from pbench.server.api.resources.query_apis import (
+    ParamType,
+    ConversionError,
+    InvalidRequestPayload,
+    MissingParameters,
+    Parameter,
+    Schema,
+)
+
+
+class TestParamType:
+    """
+    Tests on the ParamType enum
+    """
+
+    def test_enum(self):
+        assert len(ParamType.__members__) == 4
```

Comment:
Is it really useful to check that there are exactly 4 members?  If you're trying to ensure that the other tests here are updated for additions to the set, perhaps a nice failure message to that effect would be helpful.

Reply:
Yes, I suppose I could, but it's a basic reminder if someone forgets to update the tests; I'm not sure a hint message is going to be all that more helpful.

Reply:
I can just imagine someone (naming no names here, but it could be me...), seeing the assertion failing and just correcting the number without it occurring to them that they should also add to the tests below it.

I think you can just add a comma and a string to the end of the assertion, and it will print the message instead of the default assertion message; I think that will be much more useful.  The message could be something like, "New ParamType member detected, please enhance the unit tests."  (Presumably anyone _removing_ a member would have already removed it from the other tests and would understand things despite the message.)

Reply:
Not only _can_ you, but you **did**.  :-D

(Having these "conversations" separated from the code that we're reviewing, in time and space, makes it hard to stay on top of these details.  ;-) )